### PR TITLE
transport: Fix data race on loopyWriter.estdStreams in AccountCheck tests

### PR DESCRIPTION
**Pull request by easwars**
#### Root Cause:
- When ping-pong loops finished, `testFlowControlAccountCheck` accessed `st.loopy.estdStreams[id]` and `client.loopy.estdStreams[id]` directly.
- Because `loopyWriter.estdStreams` is private and unsynchronized (`loopyWriter.run(`) assumes sole ownership), reading from this map across goroutines raced with `cleanupStreamHandler` deleting streams during timeouts or stream shutdown.
 - Furthermore, when client stream goroutines encountered errors (e.g. DeadlineExceeded) during ping-pongs, wg.Wait() returned while t.Errorf() allowed the main test goroutine to continue inspecting broken streams.

#### Solution:
- Added `outStreamRequestForTesting` to `controlBuf`. This allows tests to retrieve `*outStream` pointers safely from within `loopyWriter.run()` without racing on `estdStreams`.
- Fail early on ping-pong goroutine failures.

RELEASE NOTES: none

### CODE REVIEW

**Comment by easwars** [comment]
@mbissa : We don't have an open issue for this. Probably we don't see it enough. I couldn't repro this in google3 though. But I found this in one of the GitHub actions run today when I making the changes to push the security advisory out and making the patch release.

File: internal/transport/transport_test.go [hunk 1]
```diff
@@ -1885,6 +1885,32 @@ func (s) TestAccountCheckDynamicWindowLargeMessage(t *testing.T) {
...
+		return s
+	case <-t.done:
+		return nil
+	case <-ctx.Done():
+		return nil
+	}
+}
+
+func (t *http2Client) getOutStream(ctx context.Context, id uint32) *outStream {
+	resp := make(chan *outStream, 1)
```
Comment (mbissa) [inline internal/transport/transport_test.go]: minor nit: identical lines of code - can be extracted to be reused.

File: internal/transport/transport_test.go [hunk 2]
```diff
@@ -1885,6 +1885,32 @@ func (s) TestAccountCheckDynamicWindowLargeMessage(t *testing.T) {
...
+	t.controlBuf.put(&outStreamRequestForTesting{streamID: id, resp: resp})
```
Comment (mbissa) [inline internal/transport/transport_test.go]: minor nit: do we need to handle error from put - else test falls back to returning nil from below and reports incorrect error?

**Review by mbissa — APPROVED** [review]
Approved with minor (optional) nits.

File: internal/transport/transport_test.go [hunk 3]
```diff
@@ -1885,6 +1885,32 @@ func (s) TestAccountCheckDynamicWindowLargeMessage(t *testing.T) {
...
+	select {
+	case s := <-resp:
+		return s
+	case <-t.done:
+		return nil
+	case <-ctx.Done():
+		return nil
+	}
+}
```
[the remark this answers is not in the document]
Comment (easwars) [reply internal/transport/transport_test.go]: Done.

File: internal/transport/transport_test.go (hunk 1 above)
Comment (easwars) [reply internal/transport/transport_test.go]: They are methods on two completely different types who have different fields. So, not possible to share.

File: internal/transport/transport_test.go [hunk 4]
```diff
@@ -1885,6 +1885,32 @@ func (s) TestAccountCheckDynamicWindowLargeMessage(t *testing.T) {
...
+	t.controlBuf.put(&outStreamRequestForTesting{streamID: id, resp: resp})
+	select {
+	case s := <-resp:
+		return s
+	case <-t.ctxDone:
+		return nil
+	case <-ctx.Done():
+		return nil
+	}
+}
```
[the remark this answers is not in the document]
Comment (easwars) [reply internal/transport/transport_test.go]: Done.

File: internal/transport/transport_test.go (hunk 2 above)
Comment (easwars) [reply internal/transport/transport_test.go]: Done.

Also, made the channel unbuffered, since we don't really need a buffer in there.

Also, renamed the methods to be clear that this is supposed to be only used for testing.
